### PR TITLE
Fixing: Undefined index: REMOTE_ADDR in lib/private/request.php

### PR DESCRIPTION
**Pull request by heppstux**
Hi Guys,

just having issues with cli and system cron on 8.0.2:

Undefined index: REMOTE_ADDR in /cloud/lib/private/request.php on line 64

This should fix the issue. Let me know, if I can be of any further help with this issue.

best, Stephan

### CODE REVIEW

**Comment by heppstux** [comment]
Dear Bot, of course, this contribution is under the terms of MIT-License :)

**Comment by karlitschek** [comment]
@DeepDiver1975 what do you think?

File: lib/private/request.php [hunk 1]
```diff
@@ -61,7 +61,8 @@ public static function getRequestID() {
 	 */
 	private static function isOverwriteCondition($type = '') {
 		$regex = '/' . OC_Config::getValue('overwritecondaddr', '')  . '/';
-		return $regex === '//' or preg_match($regex, $_SERVER['REMOTE_ADDR']) === 1
+		$remoteAddress = isset($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : '';
```
Comment (DeepDiver1975) [inline lib/private/request.php]: in case $_SERVER['REMOTE_ADDR'] is not set I guess we cannot do much anyhow - @LukasReschke

**Comment by DeepDiver1975** [comment]
@heppstux please open PR always against master branch first - thanks

File: lib/private/request.php (hunk 1 above)
Comment (LukasReschke) [inline lib/private/request.php]: Looks legit to me to do that when we come over a cron job.

@heppstux Can you please reopen this against master if required? THX.

**Comment by heppstux** [comment]
I didn't open this against the master-branch since request.php is gone there (see 886bda5f81d52ba4443094e4c2fffac33c27bc4b). is seems more like a patch against the "stable" version 8 branch.

I don't know if the problem is still existing on the master branch – didn't  check that. but I guess, it requires totally different pr to fix that on master.

**Comment by DeepDiver1975** [comment]
> I didn't open this against the master-branch since request.php is gone there (see 886bda5). is seems more like a patch against the "stable" version 8 branch.
> 
> I don't know if the problem is still existing on the master branch – didn't check that. but I guess, it requires totally different pr to fix that on master.

So true ... thanks a lot

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

**Comment by th3fallen** [comment]
not a whole lot to test lol but :+1:

File: lib/private/request.php [hunk 2]
```diff
@@ -61,7 +61,8 @@ public static function getRequestID() {
...
+		return $regex === '//' or preg_match($regex, $remoteAddress) === 1
```
Comment (th3fallen) [inline lib/private/request.php]: nit picky but god i hate `or` and `and` :imp: `||` , `&&` ftw

Comment (DeepDiver1975) [inline lib/private/request.php]: yes - we shall use || and  && - but this is already in the current code base as of stable7 - I see no reason to change this here

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by heppstux** [comment]
thanks

**Pull request merged**